Updates prefetch integration to add "only prefetch link on hover/mouseover/focus" option (#6585)
withastro/astro: packages/integrations/prefetch/README.md

  })],
});
```

### config.intentSelector

> By default, the prefetch script also searches the page for any links that include a `rel="prefetch-intent"` attribute, ex: `<a rel="prefetch-intent" />`. This behavior can be changed in your astro.config.* file to use a custom query selector when finding prefetch-intent links.

Reviewer: ```suggestion
By default, the prefetch script also searches the page for any links that include a `rel="prefetch-intent"` attribute, ex: `<a rel="prefetch-intent" />`. This behavior can be changed in your `astro.config.*` file to use a custom query selector when finding prefetch-intent links.
```
Never let me look at something again, or I'll find something else!  :joy:

Yes, good to go! Sorry, in general it's usually fine if you just applied my suggestions, and I don't need to review again. But, while I'm here, let's inline code the file name here!

  })],
});
```

### config.intentSelector
By default, the prefetch script also searches the page for any links that include a `rel="prefetch-intent"` attribute, ex: `<a rel="prefetch-intent" />`. This behavior can be changed in your astro.config.* file to use a custom query selector when finding prefetch-intent links.

> Importantly, the intentSelector accepts both strings and arrays, giving you the option to prefetch multiple selectors on hover if you want.

Reviewer: ```suggestion
```
I think we can handle this with strategic inline commenting instead...

__`astro.config.mjs`__

```js
import { defineConfig } from 'astro/config';
import prefetch from '@astrojs/prefetch';

export default defineConfig({
  // ...
  integrations: [prefetch({

>     // Only prefetch links with an href that begins with `/products` or `/coupons`
>     intentSelector: ["a[href^='/products']", "a[href^='/coupons']"]
> 
>     // Below is also valid syntax
>     // intentSelector: "a[href^='/products']"

Reviewer: ```suggestion
     // Only prefetch links with an href that begins with `/products` or `/coupons`
    intentSelector: ["a[href^='/products']", "a[href^='/coupons']"]
    
    // Use a string to prefetch a single selector
    // intentSelector: "a[href^='/products']"
```
I think this is a bit more illustrative than the previous, and means we can avoid a sentence!
